### PR TITLE
refactor(workspace): decouple write and addon path computation

### PR DESCRIPTION
**Pull request by Lou1415926**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/storage_init.go [hunk 1]
```diff
@@ -686,6 +687,7 @@ func (o *initStorageOpts) addonBlobs() ([]addonBlob, error) {
 			name:        o.storageName,
```
Comment (efekarakus) [inline internal/pkg/cli/storage_init.go]: nit: why not have the name be `fmt.Sprintf("%s.yml", o.storageName)`?

**Review by efekarakus — APPROVED** [review]
lgtm, just a tiny suggestion

**Review by huanjani — APPROVED** [review]
LGTM!

File: internal/pkg/cli/storage_init.go (hunk 1 above)
Comment (Lou1415926) [reply internal/pkg/cli/storage_init.go]: The intention was that perhaps we will support non-CFN addons in the future. Then the addon itself needs to decide what extension it's supposed to be. But yea, this is probably a premature generalization. Changed it to `".yml"`!

**Pull request merged**